packages/service-library/tests/fastapi/test_long_running_tasks.py

```diff
@@ -200,18 +227,83 @@ async def test_error_after_result(
     high_status_poll_interval: PositiveFloat,
 ) -> None:
     result = await async_client.post(f"{run_server}/string-list-task")
-    assert result.status_code == status.HTTP_200_OK
+    assert result.status_code == status.HTTP_202_ACCEPTED
     task_id = result.json()
 
     client = long_running_tasks.client.Client(
         app=client_app, async_client=async_client, base_url=run_server
     )
     with pytest.raises(RuntimeError):
-        async with long_running_tasks.client.periodic_task_result(
+        async with long_running_tasks.client.periodic_tasks_results(
             client,
-            task_id,
+            [task_id],
             task_timeout=10,
             status_poll_interval=high_status_poll_interval,
-        ) as string_list:
+        ) as results:
+            string_list = results[0]
             assert string_list == [f"{x}" for x in range(10)]
             raise RuntimeError("has no influence")
+
+
+class ExpectedTask(BaseModel):
+    url: str
+    params: Optional[dict[str, Any]]
+    result: Any
+
+    @property
+    def sort_key(self) -> int:
+        return id(self)
+
+
+@pytest.mark.parametrize("sort_reversed", [True, False])
+async def test_workflow_results_order(
+    run_server: AnyHttpUrl,
+    client_app: FastAPI,
+    async_client: AsyncClient,
+    high_status_poll_interval: PositiveFloat,
+    sort_reversed: bool,
+):
+    expected_tasks: list[ExpectedTask] = sorted(
+        [
+            ExpectedTask(
+                url=f"{run_server}/string-list-task",
+                params=None,
+                result=[f"{x}" for x in range(10)],
+            ),
+            ExpectedTask(
+                url=f"{run_server}/waiting-task",
+                params=dict(wait_for=ITEM_PUBLISH_SLEEP * 3),
+                result=ITEM_PUBLISH_SLEEP * 3 + 42,
+            ),
+            ExpectedTask(
+                url=f"{run_server}/waiting-task",
```

Comment:
can you explain what this is about?

Reply:
This is no longer required, as the piece of code was removed.